src/ImageSharp/Formats/Tiff/TiffFrameMetadata.cs

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Linq;
+using SixLabors.ImageSharp.Common.Helpers;
+using SixLabors.ImageSharp.Formats.Tiff.Constants;
+using SixLabors.ImageSharp.Metadata;
+using SixLabors.ImageSharp.Metadata.Profiles.Exif;
+
+namespace SixLabors.ImageSharp.Formats.Tiff
+{
+    /// <summary>
+    /// Provides Tiff specific metadata information for the frame.
+    /// </summary>
+    public class TiffFrameMetadata : IDeepCloneable
```

Comment:
@brianpopow @IldarKhayrutdinov These readonly properties are the only sticky point now. All other metadata types are read/write.

The difference I see here is that we are conflating properties that belong to a subset of metadata (EXIF) with raw metadata embedded in the format. 

For example:

With Gif frames we expose the color table length, delay, and disposal method. Those are all parsed from the `GraphicsControlExtension`.

https://github.com/SixLabors/ImageSharp/blob/a8cb71158493096121439d937b7e79aa5ea2727a/src/ImageSharp/Formats/Gif/GifFrameMetadata.cs#L34-L49

For Png we expose bit depth, color type, an interlace method and other properties. These are parsed from the header and other chunks.

https://github.com/SixLabors/ImageSharp/blob/a8cb71158493096121439d937b7e79aa5ea2727a/src/ImageSharp/Formats/Png/PngMetadata.cs#L47-L57

The same is with other formats. EXIF is treated as a separate concern. 

We have an EXIF profile embedded in each frame now, profiles should be mutable properties. 

Considering it is very likely we will come across another format which contains per frame metadata (_HEIC stores metadata for each embedded jpeg_) we should expose an EXIF property and internal XMP property to the `ImageFrameMetadata` class in the same manner we do for `ImageMetadata` (_we don't have XMP in this yet but Png would benefit from it eventually_).

https://github.com/SixLabors/ImageSharp/blob/a8cb71158493096121439d937b7e79aa5ea2727a/src/ImageSharp/Metadata/ImageMetadata.cs#L120

https://github.com/SixLabors/ImageSharp/blob/a8cb71158493096121439d937b7e79aa5ea2727a/src/ImageSharp/Metadata/ImageFrameMetadata.cs

Anything that is a member of an EXIF profile should be updated via that profile and the property removed from this type. If the EXIF profile is set to `null`, sensible defaults should be used by the encoder to cater for missing information. 

Anything that is a required metadata property that is not contained in an embedded profile should be exposed in this and the parent metadata type. If there are no properties, I would not delete this additional custom type unless we intend on implementing embedded jpeg support at some point.

Does this make sense?

Reply:
yes this makes sense. I will try to get this done in the next few days.

Reply:
Legend. Thanks mate!

Reply:
@JimBobSquarePants @brianpopow 

Sorry for the delay in answering. This method is definitely better than currently implemented. 

Really, the TIF format isn't like an exist formats. 
Those properties that are just getters of [TiffFrameMetadata](https://github.com/SixLabors/ImageSharp/blob/275a6cc27d30be50e576e540f0720eb38df10094/src/ImageSharp/Formats/Tiff/TiffFrameMetadata.c) (`Compression`, `Predictor`...), which I would call `FIleInfo` tags, are probably of interest to the user, since they look comfortable. 

The second aspect of `FIleInfo` tags - they aren't pure metadata and doesn't need (it's no use) to be edited. To create a new image with different parameters (`Compression`...) it's required to use the save method with the desired [TiffEncoder ](https://github.com/SixLabors/ImageSharp/blob/275a6cc27d30be50e576e540f0720eb38df10094/src/ImageSharp/Formats/Tiff/TiffEncoder.cs)parameters.

Reply:
@JimBobSquarePants while I was working on the changes, I realized that the `TiffFrameMetadata` is only used during decoding and there basically only to make it more convenient to access the relevant data for decoding from the ExifProfile of a frame. Because of that, I have decided to make the class internal. 

I have moved the ExifProfile and the XmpProfile to the `ImageframeMetadata` as recommended. With that the user can still access all relevant metadata via the ExifProfile.

The relevant commits are:

- 04b6f3ff: Remove ExifProfile from TiffFrameMetadata, Add ExifProfile and XmpProfile to ImageFrameMetaData
- 09f4957c: Make TiffFrameMetadata internal